lib/extensions/ar_virtual.rb

```diff
@@ -400,12 +400,15 @@ def virtual_attribute_names
       end
     end
 
-    def attributes_builder
-      @attributes_builder ||= ::ActiveRecord::AttributeSet::Builder.new(attribute_types, primary_key) do |name|
-        unless columns_hash.key?(name) || virtual_attribute?(name)
-          _default_attributes[name].dup
-        end
+    def attributes_builder # :nodoc:
+      unless defined?(@attributes_builder) && @attributes_builder
```

Comment:
What is this saying?  defined and not nil?  Why change this from `@ivar ||=` ?  It seems unrelated.

Plus, I mentally convert `unless` to `if !xxx` so `&&` requires me to think whereas the original feels easier to understand.

Reply:
This is likely because we tend to straight copy upstream and patch a part of it (so this code may actually be upstream code.